resources/js/stores/useCompare.js

```diff
@@ -0,0 +1,101 @@
+import { useLocalStorage, StorageSerializers } from '@vueuse/core';
+
+export const compare = useLocalStorage('compare', {}, {
+    serializer: StorageSerializers.object
+});
+
+export const createCompare = async function () {
+    try {
+        // get a new id for the user;
+        let response = await magentoGraphQL("mutation { createCompareList { uid item_count attributes { code label } items { uid product { sku name url_key image { url label } } attributes { code value } } } }");
+
+        compare.value = response.data.createCompareList;
+    } catch (error) {
+        Notify(window.config.translations.errors.wrong, 'error')
+        console.error(error)
+    }
+}
+
+export const refreshCompare = async function() {
+    if (!compare.value?.uid) {
+        return;
+    }
+
+    let response = await magentoGraphQL("query($uid: ID!) { compareList(uid: $uid) { uid item_count attributes { code label } items { uid product { sku name url_key image { url label } } attributes { code value } } } }", {
```

Comment:
So we don't acidentally mismatch the data i think we should extract the compareList data into it's own variable like we have done for the cart in the core
https://github.com/rapidez/core/blob/a8a399077136af19acb4edda4fdae2a8171f125e/resources/js/stores/useCart.js#L23 
https://github.com/rapidez/core/blob/a8a399077136af19acb4edda4fdae2a8171f125e/src/Http/ViewComposers/ConfigComposer.php#L45